src/jobs/java_maven_build_test.yml

```diff
@@ -0,0 +1,43 @@
+description: >
+  Build and test java app using maven; update circleCI caches and workspace.
+  Supports app-module from monorepo as well as single app from single-app-repo.
+
+parameters:
+  appname:
+    description: "Name of maven module for the app, or blank for single-app-repo"
+    type: string
+    default: ""
+  maven_cache_key_prefix:
+    description: "Prefix for the maven artifacts cache-key (used in combination with checksum of pom.xml); no caching with blank prefix"
+    type: string
+    default: ""
+  executor:
+    description: "Executor for the build (java_builder_docker is more lightweight, but some tests leveraging testcontainers etc require the java_builder_vm)"
+    type: enum
+    enum: [ "java_builder_docker", "java_builder_vm"]
+    default: "java_builder_docker"
+  java_builder_docker_java_version:
+    description: "[required for java_builder_docker] Java version for docker-executor"
+    type: string
+    default: "11.0"
+  java_builder_vm_image:
+    description: "[required for java_builder_vm] VM-image for executor"
+    type: string
+    default: "ubuntu-2004:202010-01"
+
+executor:
+  name: << parameters.executor >>
+  java_version: << parameters.java_builder_docker_java_version >>
+  vm_image: << parameters.java_builder_vm_image >>
```

Comment:
@robert-nemet do you think this pseudo-polymorph executor can work?

if i validate the orb with `circleci orb pack --skip-update-check src > orb.yml ; circleci orb --skip-update-check validate orb.yml` i notice that

if `java_builder_docker` does not specify the `vm_image` parameter, validation fails with
```
Error: Error calling job: 'java_maven_build_test'
Error calling executor: 'java_builder_docker'
Unexpected argument(s): vm_image
```
however if i add the `vm_image` parameter there (dummy param, not used), validation does _NOT_ fail for `java_builder_vm`, which i would expect to have the same problem with the `java_version` parameter that is only present on `java_builder_docker`

TLDR: im a bit afraid that the resulting orb does not work for the `java_builder_vm`, and that the validation does not find the problem because it does not dynamically resolve the executor name and rather bakes something in the first time it happens to see a value 🤔

Reply:
> Executor `java_builder_docker` do not support parameter `vm_image`.

yes i know, i wrote it 😂 
im just wondering why the other way around the problem is not found in validation - i.e. if i add the `vm_image` param to `java_builder_docker`, the validation does not realize that there is params from `java_builder_docker` that are not supported on `java_builder_vm` 😟

Reply:
i refactored this now since i think it would not work and fail at runtime (even though validation did not catch the problem)

6b19880b81b568ef74ecaaa16ed972a30b3ec9e2 introduces a custom brute-forced polymorph behaviour that is closest to what i would like to have for an easy configuration in our java repos